specification/index.bs

```diff
@@ -143,7 +143,19 @@ This key may be present.
 
 ### Key `externally_connectable`
 
-This key may be present.
+The <a href="#key-externally_connectable">`externally_connectable`</a> key declares which extensions and web pages can establish connections to the extension using [=runtime.connect()=] and [=runtime.sendMessage()=]. If omitted, all extensions may connect, but no web pages can connect.
+
+This key may be present and may include the following optional keys:
+
+#### Key `ids`
+
+A [=list=] of [=extension IDs=] that specifies which extensions can communicate with the extension. To allow all extensions, include the pattern `*` in the list. If left empty or omitted, no extensions can connect.
+
+#### Key `matches`
+
+A [=list=] of [=match patterns=] that specifies which web pages can communicate with the extension. If left empty or omitted, no web pages can connect.
```

Comment:
There is an edge case where document URLs cannot be described by a match pattern. For these cases, we specified an algorithm to determine the URL to use for matching content scripts, at [specification/index.bs: "Determine the URL for matching a document"](https://github.com/w3c/webextensions/blob/6d2086f01a4d149bcd1ec024eb40e8b08b3e2072/specification/index.bs#L301-L331).

Can we point to this?

@oliverdunk Could you confirm that this is what Chrome does (or intends to use)?

Reply:
I checked Chrome's source code for relevant behavior, which is at https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/messaging/message_service.cc;l=538-540;drc=373d7fdb08be061501a27cf72cb0ff846fbd0899

So it looks like these APIs can only be used from documents whose URL matches the match pattern. In practice, http(s): and file:-URLs. Not URLs like about:blank, blob:-, etc.

So let's resolve this thread by editing the section to emphasize that the match pattern should match the document's URL.